roachtest: cdc/sink-chaos/rangefeed=true failed

SHA: https://github.com/cockroachdb/cockroach/commits/71681f60031c101f17339236e2ba75f7a684d1a1

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=cdc/sink-chaos/rangefeed=true PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1155867&tab=buildLog

```
The test failed on master:
	cluster.go:1226,cdc.go:600,cdc.go:117,cdc.go:416,test.go:1211: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1155867-cdc-sink-chaos-rangefeed-true:4 -- ./workload fixtures load tpcc --warehouses=100 --checks=false {pgurl:2} returned:
		stderr:
		
		stdout:
		)
		I190228 15:32:17.495800 72 ccl/workloadccl/fixture.go:513  loaded warehouse (2m35s, 100 rows, 0 index entries, 5.1 KiB)
		I190228 15:33:41.915486 73 ccl/workloadccl/fixture.go:513  loaded district (3m59s, 1000 rows, 0 index entries, 99 KiB)
		I190228 15:34:00.172942 76 ccl/workloadccl/fixture.go:513  loaded order (4m17s, 3000000 rows, 6000000 index entries, 166 MiB)
		I190228 15:34:30.338193 75 ccl/workloadccl/fixture.go:513  loaded history (4m48s, 3000000 rows, 6000000 index entries, 380 MiB)
		I190228 15:34:32.635863 74 ccl/workloadccl/fixture.go:513  loaded customer (4m50s, 3000000 rows, 3000000 index entries, 1.7 GiB)
		I190228 15:34:36.040207 79 ccl/workloadccl/fixture.go:513  loaded stock (4m53s, 10000000 rows, 10000000 index entries, 3.1 GiB)
		I190228 15:34:37.989229 80 ccl/workloadccl/fixture.go:513  loaded order_line (4m55s, 29995875 rows, 29995875 index entries, 2.0 GiB)
		Error: restoring fixture: PostLoad hook: pq: count-leases: AS OF SYSTEM TIME: cannot specify timestamp in the future
		Error:  exit status 1
		: exit status 1

```

SHA: https://github.com/cockroachdb/cockroach/commits/d888b76df319571afe4d5816f1a1f0f53905653f

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=cdc/sink-chaos/rangefeed=true PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1159641&tab=buildLog

```
The test failed on master:
	cluster.go:1226,cdc.go:612,cdc.go:124,cluster.go:1564,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1159641-cdc-sink-chaos-rangefeed-true:4 -- ./workload run tpcc --warehouses=100 --duration=30m --tolerate-errors {pgurl:1-3}  returned:
		stderr:
		
		stdout:
		0.0      0.0 newOrder
		  26m55s      258            0.0            1.3      0.0      0.0      0.0      0.0 orderStatus
		  26m55s      258            0.0           13.2      0.0      0.0      0.0      0.0 payment
		  26m55s      258            0.0            1.2      0.0      0.0      0.0      0.0 stockLevel
		E190303 07:59:06.910016 1 workload/cli/run.go:419  error in payment: dial tcp 10.142.0.63:26257: connect: connection refused
		  26m56s      287            0.0            1.3      0.0      0.0      0.0      0.0 delivery
		  26m56s      287            0.0           13.2      0.0      0.0      0.0      0.0 newOrder
		  26m56s      287            0.0            1.3      0.0      0.0      0.0      0.0 orderStatus
		  26m56s      287            0.0           13.2      0.0      0.0      0.0      0.0 payment
		  26m56s      287            0.0            1.2      0.0      0.0      0.0      0.0 stockLevel
		E190303 07:59:07.986067 1 workload/cli/run.go:419  error in payment: dial tcp 10.142.0.63:26257: connect: connection refused
		: signal: killed
	cluster.go:1585,cdc.go:212,cdc.go:416,test.go:1211: dial tcp 35.231.144.244:26257: connect: connection refused

```

SHA: https://github.com/cockroachdb/cockroach/commits/de1793532332fb64fca27cafe92d2481d900a5a0

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=cdc/sink-chaos/rangefeed=true PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1160394&tab=buildLog

```
The test failed on master:
	cluster.go:1226,cdc.go:612,cdc.go:124,cluster.go:1564,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1160394-cdc-sink-chaos-rangefeed-true:4 -- ./workload run tpcc --warehouses=100 --duration=30m --tolerate-errors {pgurl:1-3}  returned:
		stderr:
		
		stdout:
		1.9      0.0      0.0      0.0      0.0 delivery
		  11m30s      196            0.0           19.2      0.0      0.0      0.0      0.0 newOrder
		  11m30s      196            0.0            1.9      0.0      0.0      0.0      0.0 orderStatus
		  11m30s      196            0.0           19.2      0.0      0.0      0.0      0.0 payment
		  11m30s      196            0.0            1.7      0.0      0.0      0.0      0.0 stockLevel
		E190304 07:43:19.088827 1 workload/cli/run.go:419  error in orderStatus: dial tcp 10.142.0.134:26257: connect: connection refused
		  11m31s      214            0.0            1.9      0.0      0.0      0.0      0.0 delivery
		  11m31s      214            0.0           19.1      0.0      0.0      0.0      0.0 newOrder
		  11m31s      214            0.0            1.9      0.0      0.0      0.0      0.0 orderStatus
		  11m31s      214            0.0           19.2      0.0      0.0      0.0      0.0 payment
		  11m31s      214            0.0            1.7      0.0      0.0      0.0      0.0 stockLevel
		: signal: killed
	cluster.go:1585,cdc.go:212,cdc.go:416,test.go:1211: unexpected node event: 2: dead

```

SHA: https://github.com/cockroachdb/cockroach/commits/e6f0a720a98c60ed0545e7cbb4b2f224b463cae9

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=cdc/sink-chaos/rangefeed=true PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1165130&tab=buildLog

```
The test failed on master:
	cluster.go:1244,cdc.go:612,cdc.go:124,cluster.go:1582,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1165130-cdc-sink-chaos-rangefeed-true:4 -- ./workload run tpcc --warehouses=100 --duration=30m --tolerate-errors {pgurl:1-3}  returned:
		stderr:
		
		stdout:
		     1.3      0.0      0.0      0.0      0.0 delivery
		  25m46s      200            0.0           12.9      0.0      0.0      0.0      0.0 newOrder
		  25m46s      200            0.0            1.3      0.0      0.0      0.0      0.0 orderStatus
		  25m46s      200            0.0           12.9      0.0      0.0      0.0      0.0 payment
		  25m46s      200            0.0            1.2      0.0      0.0      0.0      0.0 stockLevel
		  25m47s      230            0.0            1.3      0.0      0.0      0.0      0.0 delivery
		  25m47s      230            2.0           12.9 103079.2 103079.2 103079.2 103079.2 newOrder
		  25m47s      230            0.0            1.3      0.0      0.0      0.0      0.0 orderStatus
		  25m47s      230            0.0           12.9      0.0      0.0      0.0      0.0 payment
		  25m47s      230            0.0            1.2      0.0      0.0      0.0      0.0 stockLevel
		E190307 08:05:00.261249 1 workload/cli/run.go:419  error in payment: dial tcp 10.142.0.14:26257: connect: connection refused
		: signal: killed
	cluster.go:1603,cdc.go:212,cdc.go:416,test.go:1214: dial tcp 35.196.204.231:26257: connect: connection refused

```

SHA: https://github.com/cockroachdb/cockroach/commits/70e3468e7ed5fb30b10eaaf972acbb0f16099d01

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=cdc/sink-chaos/rangefeed=true PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1178890&tab=buildLog

```
The test failed on master:
	cluster.go:1267,cdc.go:625,cdc.go:125,cluster.go:1605,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1178890-cdc-sink-chaos-rangefeed-true:4 -- ./workload run tpcc --warehouses=100 --duration=30m --tolerate-errors {pgurl:1-3}  returned:
		stderr:
		
		stdout:
		     0.8      0.0      0.0      0.0      0.0 delivery
		   8m54s      168            0.0            8.5      0.0      0.0      0.0      0.0 newOrder
		   8m54s      168            1.0            0.8 103079.2 103079.2 103079.2 103079.2 orderStatus
		   8m54s      168            0.0            8.1      0.0      0.0      0.0      0.0 payment
		   8m54s      168            0.0            0.5      0.0      0.0      0.0      0.0 stockLevel
		   8m55s      186            0.0            0.8      0.0      0.0      0.0      0.0 delivery
		   8m55s      186            0.0            8.5      0.0      0.0      0.0      0.0 newOrder
		   8m55s      186            0.0            0.8      0.0      0.0      0.0      0.0 orderStatus
		   8m55s      186            1.0            8.1 103079.2 103079.2 103079.2 103079.2 payment
		   8m55s      186            0.0            0.5      0.0      0.0      0.0      0.0 stockLevel
		E190315 06:41:21.612642 1 workload/cli/run.go:420  error in payment: dial tcp 10.142.0.18:26257: connect: connection refused
		: signal: killed
	cluster.go:1626,cdc.go:213,cdc.go:417,test.go:1214: dial tcp 35.237.207.254:26257: connect: connection refused

```

SHA: https://github.com/cockroachdb/cockroach/commits/f95d45653df6be5587fb9887de241f50b6932000

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=cdc/sink-chaos/rangefeed=true PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1185396&tab=buildLog

```
The test failed on master:
	cluster.go:1267,cdc.go:625,cdc.go:125,cluster.go:1605,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1185396-cdc-sink-chaos-rangefeed-true:4 -- ./workload run tpcc --warehouses=100 --duration=30m --tolerate-errors {pgurl:1-3}  returned:
		stderr:
		
		stdout:
		      0.0      0.0 orderStatus
		   6m36s      170            0.0           11.8      0.0      0.0      0.0      0.0 payment
		   6m36s      170            0.0            0.9      0.0      0.0      0.0      0.0 stockLevel
		E190319 06:45:49.216395 1 workload/cli/run.go:420  error in stockLevel: dial tcp 10.142.0.66:26257: connect: connection refused
		_elapsed___errors__ops/sec(inst)___ops/sec(cum)__p50(ms)__p95(ms)__p99(ms)_pMax(ms)
		   6m37s      180            0.0            1.2      0.0      0.0      0.0      0.0 delivery
		   6m37s      180            0.0           11.4      0.0      0.0      0.0      0.0 newOrder
		   6m37s      180            0.0            1.2      0.0      0.0      0.0      0.0 orderStatus
		   6m37s      180            0.0           11.7      0.0      0.0      0.0      0.0 payment
		   6m37s      180            0.0            0.9      0.0      0.0      0.0      0.0 stockLevel
		E190319 06:45:50.216421 1 workload/cli/run.go:420  error in payment: dial tcp 10.142.0.66:26257: connect: connection refused
		: signal: killed
	test.go:1202: test timed out (11h49m3.061056691s)
	cluster.go:1626,cdc.go:213,cdc.go:417,test.go:1214: unexpected node event: 2: dead
	test.go:978,asm_amd64.s:523,panic.go:513,log.go:219,cluster.go:926,context.go:90,cluster.go:916,test.go:1159,asm_amd64.s:522,panic.go:397,test.go:774,test.go:760,cluster.go:1626,cdc.go:213,cdc.go:417,test.go:1214: write /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20190319-1185396/cdc/sink-chaos/rangefeed=true/test.log: file already closed

```

--- a/pkg/ccl/changefeedccl/sink.go
+++ b/pkg/ccl/changefeedccl/sink.go
@@ -405,7 +405,7 @@ func (s *kafkaSink) EmitResolvedTimestamp(
 			topics = append(topics, topic)
 		}
 		if err := s.client.RefreshMetadata(topics...); err != nil {
-			return err
+			return &retryableSinkError{cause: err}
 		}
 		s.lastMetadataRefresh = timeutil.Now()
 	}
@@ -423,7 +423,7 @@ func (s *kafkaSink) EmitResolvedTimestamp(
 		// be picked up and get later ones.
 		partitions, err := s.client.Partitions(topic)
 		if err != nil {
-			return err
+			return &retryableSinkError{cause: err}
 		}
 		for _, partition := range partitions {
 			msg := &sarama.ProducerMessage{

